assignments/nested-parens.adoc

```diff
@@ -0,0 +1,228 @@
+= Exercise: Nested Parens
+:source-language: rust
+
+In this exercise, you will learn
+
+* how to translate a `for loop` into an `iterator` solution by using `map`, `scan`, and `fold`
+* how to setup a benchmarking harness to compare speeds
+* how to use `Rayon` to parallelize your iterators!
+* how to compress data to get even more speedups
+
+We will take a famous coding problem and iterate solutions, and measure their performance.
+Problem statement: Given a well-formed set of `String` parentheses (one entry per line, all lines in a single text file), 
+
+[source, text, linenums]
+----
+()(())
+(())()
+(()())
+()()()
+((()))
+
+----
+
+find the line where there is the maximum number of nested parentheses. 
+In the above example, it would be one line `5`, as the maximum nesting level is `3`.
+
+
+In this assignment, you will be given:
+* Code to write to a local file the parenthesis inputs (which you can assume are only nested correctly)
```

Comment:
```suggestion
* Code to write the parenthesis inputs to a local file (which you can assume they are nested correctly)
```
I'm confused about this line, should it be `read` instead maybe? otherwise what do you mean when you say we an assume they are nested correctly while writing?

Reply:
Resolved above.